substrate/frame/revive/src/address.rs

```diff
@@ -100,18 +99,21 @@ where
 	T: Config<AccountId = AccountId32>,
 {
 	fn to_address(account_id: &AccountId32) -> H160 {
-		H160::from_slice(&<AccountId32 as AsRef<[u8; 32]>>::as_ref(&account_id)[..20])
+		let account_bytes: &[u8; 32] = account_id.as_ref();
+		if Self::is_eth_derived(account_id) {
+			// this was originally an eth address
+			// we just strip the 0xEE suffix to get the original address
+			H160::from_slice(&account_bytes[..20])
+		} else {
+			// this is an (ed|sr)25510 derived address
+			// avoid truncating the public key by hashing it first
+			let account_hash = blake2_256(account_bytes);
```

Comment:
blake2 is faster. Eth code will never need to deal with this hash so it is fine to not use keccak.

Reply:
It doesn't actually matter what 20 bytes you pick here, correct – you just take the trailing 20 bytes because this seems to be the standard?

Reply:
Auditors recommended to move to keccak to be consistent with other protocols. While not strictly necessary it is just out of caution. Performance should not matter as we are just hashing a few bytes.